truenas_install/__main__.py

```diff
@@ -465,14 +465,20 @@ def main():
                                                 "--efi-directory=/boot/efi",
                                                 "--bootloader-id=debian",
                                                 "--recheck",
-                                                "--no-floppy"]
-                                    if not os.path.exists("/sys/firmware/efi"):
-                                        grub_cmd.append("--no-nvram")
+                                                "--no-floppy",
+                                                "--no-nvram"]
                                     run_command(grub_cmd)
 
                                     run_command(["chroot", root, "mkdir", "-p", "/boot/efi/EFI/boot"])
                                     run_command(["chroot", root, "cp", "/boot/efi/EFI/debian/grubx64.efi",
                                                  "/boot/efi/EFI/boot/bootx64.efi"])
+
+                                    if os.path.exists("/sys/firmware/efi"):
+                                        run_command(["chroot", root, "efibootmgr", "-c",
+                                                     "-d", f"/dev/{disk}",
+                                                     "-p", f"{efi_partition_number}",
+                                                     "-L", f"TrueNAS-{disk}",
```

Comment:
Is it a good idea to put disk name, that may change into the label?  Will it add value or confusion?

Reply:
I added disk name into label to clarify which boot entry belongs to which disk. Having same Label strings is discouraged and `efibootmgr` issues a couple of warnings.

Reply:
We could to `for i, disk in reversed(enumerate(disks)):` to get a decreasing counter `i` to use in the label, that way the last boot entry we add is 0.

By the way, if middleware has something for adding or replacing boot disks after install we should make sure it does the same thing there as well.

Reply:
I have removed the disk name from label and included a counter. Let me know if we need to go reverse about it.